docker/Dockerfile_dev-ros

```diff
@@ -1,5 +1,6 @@
 # syntax=docker/dockerfile:1
-FROM ros:humble-ros-base as main-setup
+ARG ROS-VERSION
```

Comment:
Note, the canonical name for this is `ROS_DISTRO`, which is widely used. Feel free to update that across the board if you have time. 

ROS_VERSION is either 1 or 2